include/fem-dem/gls_vans.h

```diff
@@ -148,9 +148,65 @@ class GLSVANSSolver : public GLSNavierStokesSolver<dim>
   void
   read_dem();
 
+
+  /**
+   * @brief This function gives the periodic offset of the domain which is needed
+   * for the periodic boundary conditions using the QCM for void fraction.
+   *
+   * @param periodic_boundary_id The id of the periodic boundary 0
+   *
+   * @return The periodic offset
+   */
+  inline Tensor<1, dim>
+  get_periodic_offset(unsigned int periodic_boundary_id) const
+  {
+    Tensor<1, dim> offset;
+
+    // Iterating over the active cells in the triangulation
+    for (const auto &cell : (*this->triangulation).active_cell_iterators())
+      {
+        if (cell->is_locally_owned() || cell->is_ghost())
```

Comment:
Is this if necessary?

Reply:
It will crash if not locally owned or ghost